packages/babel-plugin-transform-classes/src/vanilla.js

```diff
@@ -93,6 +95,12 @@ export default class ClassTransformer {
 
     this.superName = this.node.superClass || t.identifier("Function");
     this.isDerived = !!this.node.superClass;
+
+    const { name } = this.superName;
+    this.extendsNative =
+      this.isDerived &&
+      (builtinClasses.has(name) || customBuiltins.has(name)) &&
```

Comment:
What happens when the extension of the native is indirect due to a mixin application? ie:

```js
class MyElement extends PolymerElement(HTMLElement) {}
```

Can we have a way to force the wrapping of a particular set of builtins?

Reply:
an easy work around could be to:
```js
class MyElement extends PolymerElement(class extends HTMLElement {}) {}
```

I don't think it's that simple to handle every time a constructor is passed around and in a meaningful way.

mixins also can be passed around as regular invoke.

I mean ... this is valid code too, right?
```js
const mixed = (E => () => PolymerElement(E))(window['HTMLElement']);
class MyElement extends mixed() {}
```
We need a compromise for common cases and IMO these are not necessarily based on Polymer.

Reply:
The most common case in the wild is that `HTMLElement` is extended via mixin application. Without support for that, this transform will be much less used than it could be.

This is why I suggest a way to force wrapping, possibly like:

```json
{
  "plugins": [
    ["@babel/transform-classes", {
      "builtins": [ "HTMLElement" ],
      "forceWrap": [ "HTMLElement" ],
    }]
  ]
}
```

Reply:
> The most common case in the wild is that HTMLElement is extended via mixin application. 

I'd like to see data about this. I use CE since ever and never used mixins there. It's also worth noticing that Babel supposed to be about ES, not DOM, as previously mentioned.

Every constructor is broken right now so this is already quite a big step forward.

> This is why I suggest a way to force wrapping, possibly like:

I'm a bit confused by the fact you don't want to double wrap polyfills for Custom Elements but you want to force wrap `HTMLElement` which basically means the transpiled code will wrap already that class ...

how does it work? You either force wrap, or you avoid runtime, when happens, the double wrap.

Unless you are thinking about having the polyfill before the transpiled code that uses that constructor ... is that the case?

Reply:
The polyfill would load before the compiled code, and so would wrap native HTMLElement to have a working constructor in the first place, otherwise without the polyfill both `new HTMLElement()` and `HTMLElement.apply` fail. So when the polyfill is loaded, the wrapping doesn't need to happen.

Optimally, code would be compiled to "force" wrapping of HTMLElement, even if it can't statically see a direct subclass, but the wrapping would bail if the constructor is callable.